### DISCUSSION TITLE
Temperature effect on jet fan thrust calculation in SES 4.1

### DISCUSSION THREAD

**Discussion by Yinan-Scott-Shi** [Open-SES/OpenSES#15] [Ideas]
Hi all,

It’s a fairly well known issue that SES V4.1 does not take into consideration the temperature effect on jet fan thrust.  As such, we (Vejapong Juttijudata, Conor Flemming, and Yinan Scott Shi), would like to start tackling the issue and aim to, in steps, improve the jet fan thrust calculations on the Open-SES platform.  

In true open source fashion, we would like to invite anyone to join force; we will disclose our progress notes; and all discussions will be carried out in this forum.

**Comment by Yinan-Scott-Shi** [comment]
Two equations from OMEGA2.FOR (Line 87-90, Line 256-259) are shown below.
The first equation is repeated in OMEGA4. FOR (Line 32 - 37): 
  **GAIN=QIFS(IFS)(VIFS(IFS)-Q /ALS(ILS))/ALS(ILS)**
The second equation is repeated in OMEGA5.FOR (Line 174-179):
  **GAIN=QIFS(IFS)(VIFS(IFS)-VNETRG)/ANETRG**

Variables within the two equations are:
- _**$\frac {\Delta P} {\rho}$**_= GAIN is representing total head caused by impulse fan which is pressure divided by air density
- _**$Q_j\times \mu$**_ = QIFS (IFTYP) gives the value of the volume flow rate of air of the impulse fan multiplied by the impulse fan pressure rise efficiency (ft^3/s) 
    - IFTYP is the impulse fan type number
- _**$u_j$**_ = VIFS (IFTYP) is the nozzle discharge velocity of the impulse fan
- IFS is impulse fan type interpreted from TYPLS (ILS) which is the line segment type (local variable)
-  _**$Q_t$**_ =  Q which is equal to QS(ISCT), the air volume flow rate at a cross section of a section that a train is not crossing. It is also the algebraic sum of the air volume flow rate and the train volume flow rate at a cross section of a section that a train is crossing (ft^3/s)
    - ISCT is section number
- _**$A_t$**_ = ALS(ILS) is the cross section area of a Line Segment (ft^2)
    - ILS is line segment number
- _**$u_{ann}$**_ =VNETRG is air velocity in region (local variable)  As Ewan B. Suggested below, this is the annulus velocity accounting for the presence of a train
- _**$A_{ann}$**_ =ANETRG is the region area (local variable) As Ewan B. Suggested below, this is the annulus area accounting for the presence of a train

The equations then can be displayed as: 
**$\frac {\Delta P} {\rho} = \frac {Q_j\times \mu\times(u_j-\frac{Q_t}{A_t})}{A_t} $**
**$\frac {\Delta P} {\rho} = \frac {Q_j\times \mu\times(u_j-u_{ann})}{A_{ann}} $**
These two equations are identical.


Questions:
1. Why is module OMEGA2 equations repeated in OMEGA4 and OMEGA5?
2. Is there any difference between ANETRG and ALS(ILS)?
3. Where is ANETRG defined in the program?

**Reply by ECB2020** [reply]
Scott,

Thank you for starting this thread.

**Answer to Q1, "Why is module OMEGA2 equations repeated in OMEGA4 and OMEGA5?"**

Because it made change management easier for the engineers modifying SES in the late 1970s.

- OMEGA2 is a routine written for SES v1 (in which the pressure loss calculation was isothermal).
- OMEGA4 and OMEGA5 were added in the late 1970s when they wrote a fire model into version 3.  It was easier to make as few changes to the isothermal code as possible and put similar code into new modules that handled the fire cases.  So OMEGA2 is used for all segments in non-fire simulations and for non-fire segments in fire simulations.  OMEGA4 and OMEGA5 are used for fire segments in fire simulations.

There are four situations in the SES code:

- Lines 32 to 37 of OMEGA2 calculate jet fan pressure change for a segment that does not have a train across its back end in non-fire runs.

- Lines 174 to 179 of OMEGA2 calculate jet fan pressure change for a segment that does have a train across its back end in non-fire runs.

- OMEGA4 is the routine for calculating pressure changes in fire segments that do not have trains in them during fire runs.

- OMEGA5 is the routine for calculating pressure changes in fire segments that DO have trains in them during fire runs.



**Answer to Q2, "Is there any difference between ANETRG and ALS(ILS)?"**

**ANETRG** is the annulus area around all trains that cross the back end of the segment.

**ALS(ILS)** is the area of the open tunnel in the segment.

If there is a train crossing the back end of a segment, the presence of that train affects the performance of any jet fans in that segment.

**VNETRG** is the air velocity in the annulus around all trains that cross the back end of the segment (or the air velocity in the open tunnel if there are no trains).

The equation for the first expression in OMEGA2 and the expression in OMEGA4 is
$\frac {\Delta P} {\rho} = \frac {Q_j\times \mu\times(u_j-u_t)}{A_t} $

Your equation for the second expression in OMEGA2 and the expression in OMEGA5 should be changed to
$\frac {\Delta P} {\rho} = \frac {Q_j\times \mu\times(u_j-u_{ann})}{A_{ann}} $

where (in addition to the variable names you defined)
 * $u_{ann}$ = VNETRG, the air velocity in the annulus around trains crossing the back end of the segment
 * $A_{ann}$ = ANETRG, the area in the annulus around trains crossing the back end of the segment

It is important that you understand the difference between $A_t$ and $A_{ann}$ as well as $u_t$ and $u_{ann}$ before you start writing code.

About a year ago I emailed a draft set of notes on SES to Conor that has a couple of relevant sections.  I've put the current version of those notes into my github repository for you to take a look at, because I think they might help you out.  You can find it at https://github.com/ECB2020/Hobyah/blob/master/documentation/SES-notes-v1-1.pdf

Sections 2 and 10 of that document are relevant to what you are trying to do (section 2 is about trains across the back end of segments and section 10 is about density corrections).

**Answer to Q3, "Where is ANETRG defined in the program?"**

ANETRG is initialized in line 172 of OMEGA2 and line 85 of OMEGA5.  In both cases it is initally given the area of the line segment (the open tunnel area).  Later it gets modified.

There are loops in the two routines that loop over all the trains in the line segment.  These loops subtract the area of a train from ANETRG if the train is crossing the back end of the segment.  The loops begin at line 212 in OMEGA2 and at line 125 in OMEGA5.  The place where the train area is subtracted is line 222 in OMEGA2 and line 135 in OMEGA5.

If there are trains crossing the back boundary of the segment, ANETRG will be less than ALS(ILS).  The reduced area increases the jet fan pressure rise and the increased velocity in the annulus decreases the jet fan pressure rise.

**Reply by Yinan-Scott-Shi** [reply]
Thank you very much Ewan!  This is very helpful!

**Reply by Yinan-Scott-Shi** [reply]
Hi @ECB2020,

Thank you very much for sharing your findings/notes!  If I understand section 10.2 correctly, you found the axial fan derating in SES 4.1 is done based on the average subsegment temperature within the axial fan segment.  We can use the same method for jet fans.  I'm still trying to catch up on the understanding of SES 4.1 code.  Thank you!

**Reply by ECB2020** [reply]
I'd like to suggest that you don't use the same method for jet fans, because the treatment of density corrections in line segments is different to vent segments.

A line segment with N subsegments in it has N+2 density correction factors.  There is one for each subsegment, one for the inflow k-factor at the back end and one for the inflow k-factor at the forwards end.

Jet fans in line segments have a definite physical location: they are at the back end.  Everything else in line segments uses a local correction (loss factors at the two ends, friction), so it makes sense to me to use a local correction for the jet fans too.

OMEGA4 and OMEGA5 already have density corrections for flow across the back ends of line segments; they are TADJB and TSSTAB(ILOWS).  TADJB applies to inflow into the back end, TSSTAB(ILOWS) applies to outflow from the back end (as well as to friction in the first subsegment).

I reckon that OpenSES should use the correction factor TADJB if the jet fan has a positive jet velocity and use TSSTAB(ILOWS) if the jet fan has a negative jet velocity.

Conor (who I guess must have delved into this aspect of SES based on his comments about using TSSTAB(ILOWS) ) may want to weigh in here.

**Reply by Yinan-Scott-Shi** [reply]
Thank you  very much, Ewan! 

I'll reach out to Conor to get some input from him

**Comment by Yinan-Scott-Shi** [comment]
Temperature has two effects on the equation:
$\frac {\Delta P} {\rho} = \frac {Q_j\times \mu\times(u_j-\frac{Q_t}{A_t})}{A_t} $
1. Air density is affected by the temperature, they are inversely related:
**$\rho \propto \frac{1}{T}$**
2. Tunnel air velocity is also affected by the temperature (assuming tunnel mass flow rate doesn't get affected by temperature)
**$u_t \propto T$**

Additionally, assuming jet fans are constant airflow rate devices, their nozzle velocity is not affected by density change (can someone confirm this?)

Let:
$\rho_l =$ local density
$u_t,_l =$ local tunnel velocity 
$Q_t,_l =$ local tunnel airflow rate
$T_l =$ local temperature
 
$B = \frac{T_r}{T}= \frac {\rho}{\rho_r} = \frac{u_t,_r}{u_t}= \frac{Q_t,_r}{Q_t}$

To incorporate the temperature effects into the SES jet fan thrust equation:
$\frac {\Delta P \times B} {\rho} = \frac {Q_j\times \mu\times(u_j-\frac{Q_t \times B}{A_t})}{A_t} $

Please comment/correct above statements

**Comment by Yinan-Scott-Shi** [comment]
_**Conor F.**_ has pointed out:

From OMEGA1.FOR (Line 28-31, Line 98-105), axial fans is derated by dividing the vent shaft pressure rise (SUM2) by the ratio of local absolute temperature to ambient absolute temperature (TADJST):

 **TADJST = TSSTAB(ISSLVS)**
 ** OMEGS(ISCT) = OMEGS(ISCT)+SPOOL*SUM2/TADJST**

Note: TADJST is the same as the variable B created in the last post.  

Incorporate TADJST into the equation:
 **TADJST = TSSTAB(ISSLVS)**
**GAIN=QIFS(IFS)(VIFS(IFS)-(Q)(TADJST)/ALS(ILS))/((TADJST)ALS(ILS))**
**GAIN=QIFS(IFS)(VIFS(IFS)-VNETRG(TADJST))/((TADJST)ANETRG)**

However, Conor F. has recommended to introduce following change instead for OMEGA4.FOR:
**ILOWSS = TABL3(ILS)
TADJST = TSSTAB(ILOWSS)
GAIN = QIFS(IFS)*( VIFS(IFS) - Q/ALS(ILS) )/ ALS(ILS)            
SUM2 = SUM2 + GAIN/TADJST**

This change aligns with a discussion between **_Vej J._** and **_Justin E._**: the tunnel velocity has already accounted for the temperature difference from the fire segment already.

$Q_t,_r = Q_t$

Therefore:
**$\frac {\Delta P} {\rho} = \frac {Q_j\times \mu\times(u_j-\frac{Q_t}{A_t})}{B \times A_t} $**
**$\frac {\Delta P} {\rho} = \frac {Q_j\times \mu\times(u_j-u_t)}{B \times A_t} $**

Similar update needs to be added to OMEGA5:
**ILOWSS = TABL3(ILS)
TADJST = TSSTAB(ILOWSS)
GAIN = QIFS(IFS)(VIFS(IFS)-VNETRG)/ANETRG            
SUM2 = SUM2 + GAIN/TADJST**

Additional notes from **_Conor F._**:
1. This will limit users to having only one subsegment in the impulse fan line segment 
2. The segment should be short, close to the subsegment lengths in the neighboring segments

**Reply by ECB2020** [reply]
You state that the air velocity in SES is already adjusted for density in fire runs based on a discussion between Justin E. and Vej J.  

I came to the opposite conclusion, and when I wrote my code I used the correction in two places:
- I divided the jet fan's volume flow by the density correction (reducing the thrust)
- I multiplied the air velocity upwind of the jet fan by the density correction (also reducing the thrust)

I programmed it in as
$\frac {\Delta P} {\rho} = \frac {Q_j\times \mu\ \times[u_j -  u_{ann} \times B ])} {A_{ann} \times B}$

i.e. similar to the equation that you proposed with corrections in two places.

I'd be interested to see what led you to conclude that air velocity is already adjusted for density in fire runs, and doesn't need to be multiplied by $B$.

**Reply by Yinan-Scott-Shi** [reply]
Thank you very much Ewan,  let me reach out to Vej to clarify this point.

**Comment by Yinan-Scott-Shi** [comment]
Next steps:
1. Implement the changes mentioned above, test to verify
    - Test without train in the jet fan segment
    - Test with train in the jet fan segment
2. Although in SES jet fan is considered as a pressure device without physical length (or length equal to the segment length), bi-directional jet fans will not intake at the same location when activated for different directions.
    - Allow user to have longer jet fan segment and to contain more subsegments
    - Allow user to input length of the jet fans (or to make sure user understand the length of segment = length of jet fan assembly)
    - Based on the activation direction of the jet fan, the temperature used for intake can be determined based on the intake subsegment location 
3. Portal wind is traditionally modelled using the impulse fan input.  This feature will eventually be converted to a pressure rise at the portal feature in OPEN-SES.  To ensure portal wind is not derated, there are two possible ways to implement:
    - add a user switch in the input to allow user not to derate a specific jet fan
    - code in that any portal segment (one end is an ambient node) jet fan is not to be de-rated

**Comment by conor-at-coanda** [comment]
Thanks @Yinan-Scott-Shi for alerting me to this discussion and to @ECB2020 for this insight into the jet fan model.

I agree that the density correction factor _B_ (temperature ratio) needs to go in two places as discussed above. In my tracing of the code in OMEGA4 and OMEGA5 I couldn't see any prior density adjustments which are carried into the jet fan thrust calculation. 

In the past I have implemented the density correction in a fairly simple, crude (but still correct!) manner, with a constraint of having one subsegment in the impulse fan segment. After catching up on @ECB2020's analysis I realize I also had the implicit assumption that the net flow was in the same direction as the impulse fan, and the correction worked fine for my applications.

We absolutely should have an if-condition to detect the flow direction and then select the appropriate temperature ratio - TADJB or TSSTAB(ILOWS). 

In terms of implementation, should we go with the (out-moded) arithmetic-if approach, for consistency with the rest of the code - https://en.wikipedia.org/wiki/Arithmetic_IF - or a modern Fortran if-statement - https://pages.mtu.edu/~shene/COURSES/cs201/NOTES/chap03/else-if.html ?
